### PR TITLE
Add a dockercfg_path parameter to the scheduler

### PR DESCRIPTION
**Pull request by MatMaul**
This is implemented by copying the dockercfg file in the executor work directory.

This handles issue #86.

### CODE REVIEW

File: pkg/scheduler/service/service.go [hunk 1]
```diff
@@ -224,21 +226,30 @@ func (s *SchedulerServer) AddHyperkubeFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&s.KMPath, "km_path", s.KMPath, "Location of the km executable, may be a URI or an absolute file path.")
 }
 
-// returns (downloadURI, basename(path))
 func (s *SchedulerServer) serveFrameworkArtifact(path string) (string, string) {
+	return s.serveFrameworkArtifactWithFilename(path, "")
+}
+
+// returns (downloadURI, basename(path))
```
Comment (sttts) [inline pkg/scheduler/service/service.go]: ..., basename(path) or filename if != ""

File: pkg/scheduler/service/service.go [hunk 2]
```diff
@@ -224,21 +226,30 @@ func (s *SchedulerServer) AddHyperkubeFlags(fs *pflag.FlagSet) {
...
+func (s *SchedulerServer) serveFrameworkArtifactWithFilename(path string, filename string) (string, string) {
 	serveFile := func(pattern string, filename string) {
```
Comment (sttts) [inline pkg/scheduler/service/service.go]: confusing parameter filename now that filename is also a toplevel parameter

**Comment by sttts** [comment]
two small nits, otherwise LGTM

File: pkg/scheduler/service/service.go (hunk 1 above)
Comment (MatMaul) [inline pkg/scheduler/service/service.go]: You mean compute the basename ? I was handling the case of an empty basename inside serveFrameworkArtifactWithFilename, but you're right it is better to do that here. In my next pull request I also removed the second return parameter in serveFrameworkArtifactWithFilename now that filename is systematically used (no edge case).

File: pkg/scheduler/service/service.go [hunk 3]
```diff
@@ -224,35 +226,40 @@ func (s *SchedulerServer) AddHyperkubeFlags(fs *pflag.FlagSet) {
...
+		basename = pathSplit[len(pathSplit)-1]
 	} else {
-		base = path
+		basename = path
+	}
+
+	return s.serveFrameworkArtifactWithFilename(path, basename), basename
+}
+
+// returns (downloadURI, basename(path))
```
Comment (jdef) [inline pkg/scheduler/service/service.go]: docs are now out of date, please update

Comment (MatMaul) [inline pkg/scheduler/service/service.go]: updated

**Comment by jdef** [comment]
holding off on merge due to upstream efforts

**Comment by jdef** [comment]
needs port to upstream contrib/mesos

**Comment by MatMaul** [comment]
It is already done on my side, I am currently waiting for some answers regarding  the google CLA on my company side. I will send a PR as soon as it is ok.

**Comment by MatMaul** [comment]
upstream PR : kubernetes/kubernetes#12858

**Pull request closed without merging**